rclpy/src/rclpy/timer.cpp

```diff
@@ -33,223 +34,173 @@
 namespace rclpy
 {
 void
-reset_timer(py::capsule pytimer)
+Timer::destroy()
 {
-  auto timer = static_cast<rcl_timer_t *>(
-    rclpy_handle_get_pointer_from_capsule(pytimer.ptr(), "rcl_timer_t"));
-  if (!timer) {
+  rcl_timer_.reset();
+  clock_handle_.reset();
+}
+
+Timer::Timer(
+  py::capsule pyclock, py::capsule pycontext, int64_t period_nsec)
+: clock_handle_(std::make_shared<Handle>(pyclock))
+{
+  auto clock = clock_handle_->cast<rcl_clock_t *>("rcl_clock_t");
+
+  auto context = static_cast<rcl_context_t *>(
+    rclpy_handle_get_pointer_from_capsule(pycontext.ptr(), "rcl_context_t"));
+  if (!context) {
     throw py::error_already_set();
   }
-  rcl_ret_t ret = rcl_timer_reset(timer);
+
+  // Create a client
+  rcl_timer_ = std::shared_ptr<rcl_timer_t>(
+    new rcl_timer_t,
+    [this](rcl_timer_t * timer)
```

Comment:
no need to capture `this` pointer in this lambda because it only uses the `timer` argument.